indexing-service/src/main/java/io/druid/indexing/overlord/PortFinder.java

```diff
@@ -45,7 +45,7 @@ private static boolean canBind(int portNum)
       new ServerSocket(portNum).close();
       return true;
     }
-    catch (BindException be) {
+    catch (SocketException se) {
```

Comment:
Based on this doc (https://docs.oracle.com/javase/7/docs/api/java/net/BindException.html), I don't think the catch block for BindException should be removed, this should just add a catch block for SocketException.

Reply:
Can u explain why?
This two blocks are completely equivalent and do the same
```
catch (BindException be) {
  // block 1
}
catch (SocketException se) {
  // block 2
}
```
```
catch (SocketException se) {}
```
So, why u want to separate `BindException` and `SocketException` catch blocks, if both of this could be treated as port is used and maybe any other socket error. U pointed to doc which said `Typically, the port is in use`. Typically but only it. I can say the same about `SocketException`.
Ur note would be reasonable if in block 1 u can provide more information to "RuntimeException", but u can't.

Reply:
Ah, that's fine then, didn't see that BindException was a subclass of SocketException

Reply:
Nice, let's merge it)

Reply:
merged, thanks!